futures-util/src/async_await/poll.rs

```diff
@@ -0,0 +1,32 @@
+use futures_core::{task, Future, Poll};
+use core::marker::Unpin;
+use core::mem::PinMut;
+
+/// A macro which returns the result of polling a future once within the
+/// current `async` context.
+///
+/// This macro is only usable inside of `async` functions, closures, and blocks.
```

Comment:
I might be missing some context, how or why is this macro to be used?

Reply:
These helper macros are used in @cramertj's `join!` and `select!`:
- [join.rs line 13](https://github.com/MajorBreakfast/futures-rs/blob/0ce2116a0b50355386760ba0df8df8cf11491f4c/futures-util/src/async_await/join.rs#L13)
- [select.rs line 18](https://github.com/MajorBreakfast/futures-rs/blob/0ce2116a0b50355386760ba0df8df8cf11491f4c/futures-util/src/async_await/select.rs#L18)

I extracted `poll!` and `pending!` out of `mod.rs` because `mod.rs` felt a bit unwieldy.

Reply:
@seanmonstar `poll` is used to poll a future once inside of an `async` function, block, or closure. Internally, it works by `await`ing a future which polls once and yields the result.